src/runner/IntegrationsWithOtherPackages.jl

```diff
@@ -0,0 +1,143 @@
+    file_extension = getkey(mimepairs, '.' * split(filename, '.')[end], ".txt")
+    MIME(mimepairs[file_extension])
+end
+
+export handle_websocket_message, message_channel
+"Called directly (through Distributed) from the main Pluto process"
+function handle_websocket_message(message)
+    try
+        result = on_websocket_message(Val(Symbol(message[:module_name])), message[:body])
+        if result !== nothing
+            @warn "Integrations `on_websocket_message($(Symbol(message[:module_name])))` returned a value, but is expected to return `nothing`"
+        end
+    catch ex
+        bt = stacktrace(catch_backtrace())
+        @error "Dispatching integrations websocket message failed:" message=message exception=(ex, bt)
+    end
+    nothing
+end
+const message_channel = Channel{Any}(10)
+
+"""
+Do not call yourself! Or do, but don't say I didn't warn you.
+Integrations should implement this to capture incoming websocket messages.
+I force returning nothing, because returning might give you the idea that
+the result is sent back to the client, which (right now) it isn't.
+If you want to send something back to the client, use `IntegrationsWithOtherPackages.message_channel`
+
+    function on_websocket_message(::Val{:MyModule}, body)::Nothing
+        # ...
+    end
+"""
+function on_websocket_message(module_name, body)::Nothing
+    throw(error("No websocket message handler defined for '$(module_name)'"))
+end
+
+export handle_request
+function handle_request(request)
+    try
+        on_request(Val(Symbol(request[:module_name])), request)
+    catch ex
+        bt = stacktrace(catch_backtrace())
+        @error "Dispatching integrations HTTP request failed:" request=request exception=(ex, bt)
+    end
+end
+
+"""
+Do not call yourself! Or do, but don't say I didn't warn you.
+Integrations should implement this to capture incoming http requests.
+Expects to result in a Dict with at least `:status => Int`, but could include more,
+as seen in the following example
+
+    function on_request(::Val{:MyModule}, request)
+        Dict(
+            :status => 200,
+            :headers => ["Content-Type" => "text/html"],
+            :body => "<html>Hi</html>",
+        )
+    end
+"""
+function on_request(module_name, request)::Dict{Symbol,<:Any}
+    throw(error("No http request handler defined for '$(module_name)'"))
+end
+
+"Still a bit unsure about this, but it will for now give you the (relative) path for your module requests"
+function get_base_url(module_name::Union{AbstractString, Symbol})
+    "/integrations/$(string(module_name))/$(workspace_info.notebook_id)"
```

Comment:
Can we flip these two?

/integrations/$(workspace_info.notebook_id)/$(string(module_name))